### ISSUE TITLE
Using DefaultPlugins breaks for multiple tests due to LogPlugin

### ISSUE THREAD

**Issue by MJohnson459** [bevyengine/bevy#1969]
## Bevy version

0.5

## Operating system & version

Ubuntu 18.04

## What you did

I am trying to write tests for systems and encounter this error when there are multiple tests

```rust
#[cfg(test)]
mod test_example {
    use super::*;

    fn setup() -> App {
        let mut builder = App::build();
        builder.add_plugins(DefaultPlugins);
        builder.app
    }

    #[test]
    fn test_one() {
        let _app = setup();
        assert!(true);
    }

    #[test]
    fn test_two() {
        let _app = setup();
        assert!(true);
    }
}
```

## What you expected to happen

Log instances shouldn't conflict

## What actually happened

> Could not set global default tracing subscriber. If you've already set up a tracing subscriber, please disable LogPlugin from Bevy's DefaultPlugins

## Additional information

A possible fix might be to remove the explicit panics if creating a global subscriber fails
https://github.com/bevyengine/bevy/blob/4a477e762dbe881430385559456b4140881b6f92/crates/bevy_log/src/lib.rs#L79

**Comment by alice-i-cecile** [comment]
Fully testing the AppBuilder like this is hard; see #1720. You may be interested in #1714, which provides an alternative strategy.

**Comment by MJohnson459** [comment]
Just to add some context for my specific problem, I'm using bevy headless with `default-features = false` so most of the threading or rendering problems (e.g. #1720 ) don't affect me. I'm also trying to test functions related to the `bevy_rapier2d` plugin which requires `app.add_plugins(RapierPhysicsPlugin)` - this makes it difficult to use `world` directly as in #1714 because each resource/system would need to be copied from the plugin manually.
(discussed on discord https://discord.com/channels/691052431525675048/742569353878437978/833666064063135766)

In other words, my use case is pretty niche. Normally I would just ignore this issue because of that but it does seem like this could be a relatively low-hanging fix to remove one of the "global" systems and make this case supported with `default-features = false`.

---

Note that I do have a work around by copying from the DefaultPlugin minus the LogPlugin so this is not in any way critical: 
```rust
        builder
            .add_plugin(CorePlugin::default())
            .add_plugin(TransformPlugin::default())
            // ... etc
            ;
```

---

Finally, prior to switching to bevy I was also using tracing and rather than setting the global subscriber directly I was using something like this:
```rust
let _ = tracing_subscriber::fmt::fmt()
    .with_max_level(Level::DEBUG)
    .try_init();
```
This assumes that the failure is always because the global subscriber already exists which the documentation just calls "likely"

> Returns an Error if the initialization was unsuccessful, likely because a global subscriber was already installed by another call to try_init.

**Comment by mockersf** [comment]
Tracing integration in Bevy is not ideal, as we are not following the normal path: we should just link and use the log macros as said in https://tracing.rs/tracing/index.html#in-libraries. We are actually setting up tracing completely and in an opinionated way...

for your specific use case, you can add the `DefaultPlugins` with a twist (didn't check the exact syntax though):
```rust
   builder.add_plugins_with(DefaultPlugins, |group| {
        group.disable::<bevy::log::LogPlugin>()
    });
```

**Comment by mockersf** [comment]
would the doc added in #1973 be good enough for you to avoid this issue?

**Comment by MJohnson459** [comment]
I'm not sure. On one hand the extra documentation is great but on the other removing the LogPlugin still forces users to add their own log handler for tests (or forgo output). I think ideally I would prefer the LogPlugin to survive being used in tests but I can also see the case for just disabling it.

I am happy for this to be closed if the other option isn't something that would be desired by others.

**Comment by Bauxitedev** [comment]
I've got a related issue I think. Whenever I call `info!` or any of the logging macros provided by `bevy::prelude` in my tests, the output doesn't show up when I run `cargo test` at all (even with the `--no-capture` flag). It works when I use `log`'s macros directly, but I don't want to add that to all my files, because then I run the risk of accidentally calling bevy's logger macros instead of log's logger macros, which means the output silently doesn't get printed during tests. Any ideas?

**Comment by alice-i-cecile** [comment]
Fixed by #6411.
